From 5cdfcfaff35233b2afd0049044257791490a50ee Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 30 Nov 2021 08:25:05 +0200
Subject: [PATCH] [SW-66047] gaudi2: sec: update check for binning

Currently binning information is received through LKD. Ideally in a
secured setup this should be ignored, instead values would be read
from efuse.

But this precondition check would fail if efuse is not burned with
binning values and security is enabled.

Hence this 'revert me' patch is required to disable this check.

Change-Id: I4b0d35fdc477abc815612efc207fed0ea72b4ef5
---

diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 7b7e1fe..0e40d67 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -129,10 +129,12 @@
 		return -1;
 	}
 
-	if (preboot_security_is_enabled()) {
-		preboot_puts("binning_conf is not supported - due to security\n");
-		return -1;
-	}
+	/*
+	 * TODO:
+	 * Currently binning information is received through LKD.
+	 * Add a security enable precondition once the binning info
+	 * is read from a valid efuse content.
+	 */
 
 	masks.tpc = mcomms->tpc_binning_conf;
 	masks.dec = mcomms->dec_binning_conf;
